src/main/java/com/powsybl/openloadflow/ac/AcloadFlowEngine.java

```diff
@@ -128,6 +128,25 @@ public AcLoadFlowResult run() {
         voltageInitializer.prepare(context.getNetwork());
 
         RunningContext runningContext = new RunningContext();
+        double distributedActivePower = 0.0;
+        ReportNode reportNode = context.getNetwork().getReportNode();
+
+        // Verify whether a regulated bus voltage exists.
+        // If not, then fail immediately with SOLVER_FAILED status.
+        // Note that this approach is not perfect and could be improved in the future, in particular in
+        // security analysis and remedial actions context.
+        // For example: a contingency may cause the last PV node to disappear. In this case here we would
+        // just report SOLVER_FAILED. However, there could be other generators blocked at MinQ or MaxQ that
+        // _could potentially_ recover the situation, but this will not be tried at all...
+        boolean hasVoltageRegulatedBus = context.getNetwork().getBuses().stream()
+                .anyMatch(b -> b.isGeneratorVoltageControlEnabled() && !b.isDisabled() && !b.getGeneratorVoltageControl().orElseThrow().isDisabled());
```

Comment:
Isn't it dangerous to have a `orElseThrow()` here?

The `AbstractLfBus.setGeneratorVoltageControlEnabled(boolean)` method doesn't ensure that the generator voltage control is present when it is called with `true`.

Maybe something like this would be more robust:
```java
boolean hasVoltageRegulatedBus = context.getNetwork().getBuses().stream()
        .anyMatch(b -> b.isGeneratorVoltageControlEnabled() && !b.isDisabled()
                && !b.getGeneratorVoltageControl().map(GeneratorVoltageControl::isDisabled).orElseGet(() -> {
                    LOGGER.error("Bus {} is said to have a generator voltage control enabled, but has no generator voltage control", b.getId());
                    ... // Report the error in reportNode 
                    return Boolean.TRUE;
                }));
```

Reply:
OK, you are right. I haven't seen that `hasGeneratorVoltageControllerCapability()` (called in the setter) checks the presence of a generator voltage control.